dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/AccessService.java

```diff
@@ -0,0 +1,62 @@
+package edu.harvard.iq.dataverse.api;
+
+import edu.harvard.iq.dataverse.PermissionServiceBean;
+import edu.harvard.iq.dataverse.persistence.datafile.datavariable.DataVariable;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.user.Permission;
+import edu.harvard.iq.dataverse.persistence.user.User;
+import io.vavr.control.Try;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Stateless
+public class AccessService extends AbstractApiBean {
```

Comment:
I'm not sure about AccessService since it indicates that it controls all access, but it only controls Embargo access, so maybe something like EmbargoAccessService?

Reply:
I don't think this class should extend AbstarctApiBean, embargo has nothing to do with api it just used there at the moment. But in terms of relations it is not related to it.

Reply:
So you can remove findDataVariable from api and just use VariableServiceBean here.

Reply:
It only controls embargo access because we didn't refactor `Access` properly yet. Making it `EmbargoAccessService` might suggest that we want separate service just for embargo related operation which is not true.

Reply:
Ok, but we also have no guarantee that we will ever refactor Access and you are browsing code in present time, changing class name would be trivial anyway since it is just one button to change it's name in all injecting places.

Reply:
We extend `AbstractApiBean` because we create dataverse requests as part of the logic.
We can discuss this on a meeting but IMO this is an api bean (with somewhat limited usability for now) so it should extend `AbstractApiBean`

Reply:
Having `getApiTokenUser` method in the class makes it not embargo explicit. 
However it's true that for now we only use it in embargo context.
Let's discuss this on a meeting since IMO we should make operation-specific classes even just in naming.

Reply:
But DataverseRequest by itself is not made only for api but also for normal session so it is also not made exclusively for APi. Yes you create dataverse requests but that doesn't mean that the object is related to dataverse request by any means. For example if human is using a car it doesn't mean that we extend the car object. We can also talk about it during meeting since it's soon :P.